src/hashtable.py

```diff
@@ -44,47 +44,62 @@ def _hash_mod(self, key):
 
 
     def insert(self, key, value):
-        '''
-        Store the value with the given key.
-
-        Hash collisions should be handled with Linked List Chaining.
-
-        Fill this in.
-        '''
-        pass
-
-
+        #put value in at random index
+        if not self.storage[self._hash_mod(key)]:
+            self.storage[self._hash_mod(key)] = LinkedPair(key, value)
+        #add to head of linked list
+        else:
+            #make new linked Pair - adding (key, value) to head
+            new_pair = LinkedPair(key, value)
+            #new_pair's next points to old value
```

Comment:
This is passing the tests, but you're not actually overwriting the old keys with new values.  Adding the new node at the head just ensures that the retrieve method will get the newest value first.
This way works, but it's a little sloppy as far as memory management.